bug: Getting started fails to build

- [x] I have looked for existing issues (including closed) about this

## Bug Report
<!--
A clear and concise description of what the bug is.
--->
In the documentation https://docs.rig.rs/docs/quickstart/getting_started there is a simple example which doesn't build with the following:
```bash
error[E0599]: no function or associated item named `from_env` found for struct `Client<Ext, H>` in the current scope
   --> src/main.rs:29:41
    |
 29 |     let openai_client = openai::Client::from_env();
    |                                         ^^^^^^^^ function or associated item not found in `Client<OpenAIResponsesExt, _>`
    |
note: if you're trying to build a new `Client<OpenAIResponsesExt, _>`, consider using `Client::<Ext, H>::new` which returns `Result<Client<_, _>, rig::http_client::Error>`
   --> /Users/andy/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rig-core-0.28.0/src/client/mod.rs:252:5
    |
252 |     pub fn new(api_key: impl Into<Key>) -> http_client::Result<Self> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: items from traits can only be used if the trait is in scope
help: there is an associated function `from` with a similar name
   --> /Users/andy/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/convert/mod.rs:592:5
    |
592 |     fn from(value: T) -> Self;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
help: trait `ProviderClient` which provides `from_env` is implemented but not in scope; perhaps you want to import it
    |
 23 + use rig::client::ProviderClient;
    |

error[E0282]: type annotations needed
  --> src/main.rs:34:20
   |
34 |       let response = gpt4
   |  ____________________^
35 | |         .prompt("Who are you?")
36 | |         .await
   | |______________^ cannot infer type

Some errors have detailed explanations: E0282, E0599.
For more information about an error, try `rustc --explain E0282`.
error: could not compile `rag-my-pdf` (bin "rag-my-pdf") due to 2 previous errors
````

## Reproduction
<!--
Code snippet.
--->

## Expected behavior
<!--
A clear and concise description of what you expected to happen.
--->
Builds

## Screenshots
<!--
If applicable, add screenshots to help explain your problem.
--->

## Additional context
<!--
Add any other context about the problem here.
--->

--- a/README.md
+++ b/README.md
@@ -89,23 +89,27 @@ cargo add rig-core
 
 ### Simple example
 ```rust
-use rig::{client::CompletionClient, completion::Prompt, providers::openai};
+use rig::client::{CompletionClient, ProviderClient};
+use rig::completion::Prompt;
+use rig::providers::openai;
 
 #[tokio::main]
-async fn main() {
-    // Create OpenAI client and model
-    // This requires the `OPENAI_API_KEY` environment variable to be set.
-    let openai_client = openai::Client::from_env();
+async fn main() -> Result<(), anyhow::Error> {
+    // Create OpenAI client
+    let client = openai::Client::from_env();
 
-    let gpt4 = openai_client.agent("gpt-4").build();
+    // Create agent with a single context prompt
+    let comedian_agent = client
+        .agent(openai::GPT_4O)
+        .preamble("You are a comedian here to entertain the user using humour and jokes.")
+        .build();
 
-    // Prompt the model and print its response
-    let response = gpt4
-        .prompt("Who are you?")
-        .await
-        .expect("Failed to prompt GPT-4");
+    // Prompt the agent and print the response
+    let response = comedian_agent.prompt("Entertain me!").await?;
 
-    println!("GPT-4: {response}");
+    println!("{response}");
+
+    Ok(())
 }
 ```
 Note using `#[tokio::main]` requires you enable tokio's `macros` and `rt-multi-thread` features

--- a/rig/rig-core/examples/agent.rs
+++ b/rig/rig-core/examples/agent.rs
@@ -1,12 +1,11 @@
-use rig::prelude::*;
-
+use rig::client::{CompletionClient, ProviderClient};
+use rig::completion::Prompt;
 use rig::providers::openai;
-use rig::{completion::Prompt, providers};
 
 #[tokio::main]
 async fn main() -> Result<(), anyhow::Error> {
     // Create OpenAI client
-    let client = providers::openai::Client::from_env();
+    let client = openai::Client::from_env();
 
     // Create agent with a single context prompt
     let comedian_agent = client

